projecttacoma/deqm-test-server-frontend
# Select autocomplete

## Summary
Implementation of reusable autocomplete component, which is currently used to select a practitioner for the evaluate measure page. When the user clicks on the autocomplete textbox, a list of up to 10 practitioners display and narrow down based on what the user types. 
## New behavior
On the Evaluate Measure page for a valid measure, an autocomplete component for a practitioner now appears. This enables the user to select an existing practitioner from the dropdown menu, which offers more specific options as the user types. _Note that the screen currently displays which option has been selected beneath the autocomplete component. This is to keep track of the state variable for later use._
## Code changes
- **pages/[resource]/[id]/SelectComponent**  defines an autocomplete textbox that provides a dropdown menu of all the resources of a given resourceType matching what the user types. 
- **pages/[resource]/[id]/evaluate.tsx**  renders autocomplete 
- **__tests__/pages/resource/id/evaluate.test.tsx**, unit tests added to file
-**__tests__/components/SelectComponent.test.tsx** unit tests added to file

## Testing Guidance
See the README.md for first time setup instructions.
#### Testing in browser: 
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
- Click on “Measure” from the left-hand-side navigation menu. Click on any of the IDs. Click on the “Evaluate Measure” button near the top right of the main page body.
- To test when no practitioners exist
  - First test the server when no practitioners exist. In this case, the webpage should display "No resources of type Practitioner found"
- To test when a practitioner does exist
  - Start by adding two new practitioners by clicking on the practitioner tab, then clicking the create new Practitioner button, then inputting a valid FHIR practitioner resource, and finally clicking submit. Return to the Evaluate Measure page using the steps above. Then complete the following three checks:
  - click on the textbox and verify that both resources display as part of a drop down menu
  - begin typing out one of the unique practitioner IDs and verify that the drop down menu now only displays the matching practitioner 
  - type something not contained in either practitioner ID to verify that the dropdown menu disappears where there are no matches. 
#### Unit testing: 
- Run the unit tests: `npm run test`
